addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_38/A5_38_08_Dimming.java

```diff
@@ -44,31 +46,53 @@ public A5_38_08_Dimming(ERP1Message packet) {
     @Override
     protected void convertFromCommandImpl(String channelId, String channelTypeId, Command outputCommand,
             State currentState, Configuration config) {
+        switch (channelTypeId) {
+            case CHANNEL_DIMMER:
+                byte dimmValue;
 
-        EnOceanChannelDimmerConfig c = config.as(EnOceanChannelDimmerConfig.class);
-        byte rampingTime = (c.rampingTime == null) ? Zero : c.rampingTime.byteValue();
-
-        if (outputCommand instanceof DecimalType) {
-            if (((DecimalType) outputCommand).equals(DecimalType.ZERO)) {
-                setData(CommandId, Zero, rampingTime, (byte) (TeachInBit | SwitchOff));
-            } else {
-                setData(CommandId, ((DecimalType) outputCommand).byteValue(), rampingTime,
-                        (byte) (TeachInBit | SwitchOn));
-            }
-        } else if ((OnOffType) outputCommand == OnOffType.ON) {
-            setData(CommandId, Switch100Percent, rampingTime, (byte) (TeachInBit | SwitchOn));
-        } else {
-            setData(CommandId, Zero, rampingTime, (byte) (TeachInBit | SwitchOff));
+                if (outputCommand instanceof DecimalType) {
+                    dimmValue = ((DecimalType) outputCommand).byteValue();
+                } else if (outputCommand instanceof OnOffType) {
+                    dimmValue = ((OnOffType) outputCommand == OnOffType.ON) ? Switch100Percent : Zero;
+                } else {
+                    throw new IllegalArgumentException(outputCommand.toFullString() + " is no valid dimming command.");
+                }
+
+                EnOceanChannelDimmerConfig c = config.as(EnOceanChannelDimmerConfig.class);
+
+                boolean eltakoDimmer = (c.eltakoDimmer == null) ? true : c.eltakoDimmer;
+                boolean storeValue = (c.storeValue == null) ? false : c.storeValue;
+
+                byte storeByte = 0x00;
+
+                if (!eltakoDimmer) {
```

Comment:
Please add comments.
Reviewers don't necessarily know about the quirks and fellow developer might not as well.

Reply:
Thanks, added some explanation for the class, and some inline comments.